magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/update_MAGIC_database.py

```diff
@@ -0,0 +1,322 @@
+                combined_df[column]
+                .fillna(default_value)
+                .astype(type(not_null_data.iloc[0]))
+            )
+
+        combined_df = combined_df.infer_objects()
+
+    return combined_df
+
+def main():
+
+    """Main function."""
+    
+    parser = argparse.ArgumentParser()
+    
+    date_min_default = "20191101"
+    current_datetime = datetime.now()
+    date_max_default = current_datetime.strftime("%Y%m%d")
+
+    parser.add_argument(
+        "--date-min",
+        "-m",
+        dest="date_min",
+        type=str,
+        default=date_min_default,
+        help="Start of the time interval (in LST convention, format YYYYMMDD)."
+    )
+
+    parser.add_argument(
+        "--date-max",
+        "-M",
+        dest="date_max",
+        type=str,
+        default=date_max_default,
+        help="End of the time interval (in LST convention, format YYYYMMDD)."
+    )
+
+    args = parser.parse_args()
+
+    config = resource_file("database_config.yaml")
+
+    with open(
+        config, "rb"
+    ) as bf:
+        config_dict = yaml.safe_load(bf)
+
+    df_path = config_dict["database_paths"]["MAGIC+LST1"]
+    df_key = config_dict["database_keys"]["MAGIC+LST1"]
+    df = pd.read_hdf(df_path, key=df_key)
+
+    #Set "" to generate a new database.
+    previous_database_path = config_dict["database_paths"]["MAGIC"]
+
+    tel_id = [1, 2]
+    roman_numerals = {
+            1: "I",
+            2: "II"
+    }
+
+    file_exists = os.path.exists(previous_database_path)
+
+    if file_exists:
+
+        new_h5_file_path = (
+            "/fefs/aswg/workspace/joanna.wojtowicz/data/Common_MAGIC_LST1_data_MAGIC_runs_subruns_UPDATED.h5"
+        )
+
+        date_min = args.date_min
+        date_max = args.date_max
+
+        print("Updating database...")
+
+        database = table_magic_runs(df, date_min, date_max)
+
+        for tel in tel_id:
+
+            tel_roman = roman_numerals[tel]
+            key = config_dict["database_keys"][f"MAGIC-{tel_roman}"]
+
+            DF = pd.read_hdf(
+                previous_database_path,
+                key=key,
+            )
+
+            updated_df = update_tables(database, DF, tel)
+            print(updated_df)
+
+        try:
+            updated_df.to_hdf(new_h5_file_path, key=key, mode=("w" if tel == 1 else "a"), format = "table")
+            print(f"File saved successfully at {new_h5_file_path}")
+
+        except Exception as e:
+            print(f"An error occurred: {e}")
+
+    else:
+        print("Database does not exist. Creating a new database...")
+        
+        new_database_file_path = "/fefs/aswg/workspace/joanna.wojtowicz/data/Common_MAGIC_LST1_data_MAGIC_runs_subruns_NEW.h5"
+
+        database_default = table_magic_runs(df, date_min_default, date_max_default)
+
+        for tel in tel_id:
+            
+            tel_roman = roman_numerals[tel]
+            key = config_dict["database_keys"][f"MAGIC-{tel_roman}"]
```

Comment:
in other parts of the code if tel_id was 1 or 2 for MAGIC-I or MAGIC-II I was using:
```"MAGIC-" + tel_id * "I"```
your version is more direct but have some extra lines of code. You can either keep it as it is or change it, as you prefer